### PR TITLE
add a gracetime and ignore nodes that have no ohai data to prevent false positives during initial convergence

### PR DESCRIPTION
**Pull request by majormoses**
Basically when a node is first spun up it checks the last convergence time which will be the epoch until it completes. To combat this we only check nodes that have had a certain amount of uptime as the data they return is unreliable otherwise.

Signed-off-by: Ben Abrams <me@benabrams.it>

#### General

- [ ] Update Changelog following the conventions laid out on [Keep A Changelog](http://keepachangelog.com/)

- [ ] Update README with any necessary configuration snippets

- [ ] Binstubs are created if needed

- [ ] RuboCop passes

- [ ] Existing tests pass 

#### Purpose
Basically when a node is first spun up it checks the last convergence time which will be the epoch until it completes. To combat this we only check nodes that have had a certain amount of uptime as the data they return is unreliable otherwise.

#### Known Compatibility Issues
None, though this is a breaking change. See changelog an purpose

### CODE REVIEW

**Comment by majormoses** [comment]
```
$ bundle exec ./bin/check-chef-nodes.rb  -U https://chef-dev.bravo.cloudcruiser.com/organizations/dev -C babrams -K ~/.chef/babrams.pem 
TRUNCATED_WARNINGS
ChefNodesStatusChecker CRITICAL: The following nodes cannot be provisioned: ip-10-55-0-114.us-west-2.compute.internal
```

**Comment by majormoses** [comment]
test with failure:
```
$ bundle exec ./bin/check-chef-nodes.rb  -U https://chef-<REDACTED>/organizations/<REDACTED> -C <REDACTED> -K ~/.chef/<REDACTED>.pem  -t 120 | egrep -v 'W, '
ChefNodesStatusChecker CRITICAL: The following nodes cannot be provisioned: ["ip-10-55-250-70.us-west-2.compute.internal", "ip-10-55-151-148.us-west-2.compute.internal", "ip-10-55-21-239.us-west-2.compute.internal", "ip-10-55-251-239.us-west-2.compute.internal", "ip-10-55-140-37.us-west-2.compute.internal", "ip-10-55-140-175.us-west-2.compute.internal", "ip-10-55-141-197.us-west-2.compute.internal", "ip-10-55-152-25.us-west-2.compute.internal", "ip-10-55-141-254.us-west-2.compute.internal", "ip-10-55-20-222.us-west-2.compute.internal", "ip-10-55-141-56.us-west-2.compute.internal", "ip-10-55-151-91.us-west-2.compute.internal", "ip-10-55-140-190.us-west-2.compute.internal", "ip-10-55-250-117.us-west-2.compute.internal", "ip-10-55-151-183.us-west-2.compute.internal", "ip-10-55-150-185.us-west-2.compute.internal", "ip-10-55-140-180.us-west-2.compute.internal", "ip-10-55-152-135.us-west-2.compute.internal", "ip-10-55-142-219.us-west-2.compute.internal", "ip-10-55-140-19.us-west-2.compute.internal", "ip-10-55-250-173.us-west-2.compute.internal", "ip-10-55-20-143.us-west-2.compute.internal", "ip-10-55-252-184.us-west-2.compute.internal", "ip-10-55-141-224.us-west-2.compute.internal", "ip-10-55-150-6.us-west-2.compute.internal", "ip-10-55-140-45.us-west-2.compute.internal", "ip-10-55-142-19.us-west-2.compute.internal", "ip-10-55-250-114.us-west-2.compute.internal", "ip-10-55-150-42.us-west-2.compute.internal", "ip-10-55-142-104.us-west-2.compute.internal", "ip-10-55-152-86.us-west-2.compute.internal", "ip-10-55-140-91.us-west-2.compute.internal", "ip-10-55-252-221.us-west-2.compute.internal", "ip-10-55-150-116.us-west-2.compute.internal", "ip-10-55-251-21.us-west-2.compute.internal", "ip-10-55-250-168.us-west-2.compute.internal", "ip-10-55-151-174.us-west-2.compute.internal", "ip-10-55-142-12.us-west-2.compute.internal", "ip-10-55-0-114.us-west-2.compute.internal", "ip-10-55-152-28.us-west-2.compute.internal", "ip-10-55-141-125.us-west-2.compute.internal"]
```

test without failure:
```
$ bundle exec ./bin/check-chef-nodes.rb  -U https://<REDACTED>/organizations/dev -C <REDACTED> -K ~/.chef/<REDACTED>.pem | egrep -v 'W, 'ChefNodesStatusChecker OK: Chef Server API is ok, all nodes reporting
```

I did a watch while a node was converging and did not see it alert.

File: bin/check-chef-nodes.rb [hunk 1]
```diff
@@ -85,21 +93,37 @@ def connection
...
+        next
+      end
+
+      # won't check if node's uptime is still within grace period
+      unless node['automatic']['uptime_seconds'] > config[:grace_period]
+        checked_nodes << { node['name'] => false }
+        next
       end
+
+      # compute elapse time since last convergence
```
Comment (eheydrick) [inline bin/check-chef-nodes.rb]: elapsed

**Review by eheydrick — APPROVED** [review]
One very minor nitpick, otherwise 👍

**Comment by majormoses** [comment]
released: https://rubygems.org/gems/sensu-plugins-chef/versions/4.0.0
